### PR TITLE
Fix generation min length

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

This PR fixes a typo in `generation_utils.py`. `min_length` of `0` has no effect so we don't need to check whether it's bigger than -1, but whether it's bigger than 0.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by Narsil — APPROVED** [review]
LGTM.

If I understand correctly we were almost always adding the min_length logits processor even though it's not needed when min length is zero right ?
(curiosity Can we generate zero-tokens actually ?)

**Comment by patrickvonplaten** [comment]
> LGTM.
> 
> If I understand correctly we were almost always adding the min_length logits processor even though it's not needed when min length is zero right ? (curiosity Can we generate zero-tokens actually ?)

Yeah that was wrong before - we shouldn't use the min_length logits processor if min_length=0
